src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/HashProviderDispenser.OSX.cs

```diff
@@ -193,20 +214,13 @@ internal AppleDigestProvider(Interop.AppleCrypto.PAL_HashAlgorithm algorithm)
                 HashSizeInBytes = hashSizeInBytes;
             }
 
-            public override unsafe void AppendHashDataCore(byte[] data, int offset, int count)
+            public override unsafe void AppendHashData(ReadOnlySpan<byte> data)
             {
-                Debug.Assert(data != null);
-                Debug.Assert(offset >= 0);
-                Debug.Assert(offset < data.Length);
-                Debug.Assert(count >= 0);
-                Debug.Assert(data.Length - offset > count);
-
                 int ret;
 
-                fixed (byte* pData = data)
+                fixed (byte* pData = &data.DangerousGetPinnableReference())
                 {
-                    byte* pbData = pData + offset;
-                    ret = Interop.AppleCrypto.DigestUpdate(_ctx, pbData, count);
+                    ret = Interop.AppleCrypto.DigestUpdate(_ctx, pData, data.Length);
```

Comment:
> If those plans are fairly firm

I don't know if or when it'll actually happen.  But I'm fine moving that into the interop file.  I'll do so in a subsequent PR.